### PR TITLE
CP-39818: Auto-launch app after installation is finalized

### PR DESCRIPTION
**Pull request by danilo-delbusso**
Example:

https://user-images.githubusercontent.com/32554698/172155939-48d66a4c-2bc8-4384-bd92-73f3e816311d.mp4

The background of the checkbox cannot be changed without worsening UX:

see [issue in GitHub](https://github.com/wixtoolset/issues/issues/1687)
or [thread here](https://wix-users.narkive.com/Ss5GWl5A/checkbox-drawn-with-different-coloured-background)
or [StackOverflow thread](https://stackoverflow.com/questions/31178681/how-can-i-make-the-background-of-an-checkbox-transparent-in-wix)

### CODE REVIEW

**Comment by danilo-delbusso** [comment]
Flagging as ready for review since this is a valid solution, and the checkbox background can be fixed by changing the banner bitmap

File: WixInstaller/XenCenter.wxs [hunk 1]
```diff
@@ -271,6 +271,20 @@
...
+			</Publish>
+         </UI>
+        <Property Id="WIXUI_EXITDIALOGOPTIONALCHECKBOXTEXT" Value="Launch $(var.BrandProduct)" />
+        <Property Id="WIXUI_EXITDIALOGOPTIONALCHECKBOX" Value="1" />
+        <Property Id="WixShellExecTarget" Value="[#XenCenterEXE]" />
+        <CustomAction Id="LaunchApplication" 
+            BinaryKey="WixCA" 
+            DllEntry="WixShellExec"
+            Impersonate="yes" />
+		<Icon Id="XenCenterICO" SourceFile="$(env.RepoRoot)\Branding\Images\AppIcon.ico" />
```
Comment (kc284) [inline WixInstaller/XenCenter.wxs:288]: Indentation.

File: WixInstaller/XenCenter.wxs [hunk 2]
```diff
@@ -271,6 +271,20 @@
             <Custom Action="ClearAllUsers" After="FindRelatedProducts">NOT Installed AND PERMACHINEINSTALL</Custom>
             <ExecuteAction Sequence="1102" />
         </InstallUISequence>
-        <Icon Id="XenCenterICO" SourceFile="$(env.RepoRoot)\Branding\Images\AppIcon.ico" />
+         <UI>
+            <Publish Dialog="ExitDialog"
```
Comment (kc284) [inline WixInstaller/XenCenter.wxs:275]: When the application in already installed and the user wants to repair, the checkbox does not appear. I think we should show it in this case too.

File: WixInstaller/XenCenter.wxs [hunk 3]
```diff
@@ -271,6 +271,20 @@
...
+                Value="LaunchApplication">WIXUI_EXITDIALOGOPTIONALCHECKBOX = 1 and NOT Installed
+			</Publish>
+         </UI>
+        <Property Id="WIXUI_EXITDIALOGOPTIONALCHECKBOXTEXT" Value="Launch $(var.BrandProduct)" />
+        <Property Id="WIXUI_EXITDIALOGOPTIONALCHECKBOX" Value="1" />
+        <Property Id="WixShellExecTarget" Value="[#XenCenterEXE]" />
+        <CustomAction Id="LaunchApplication" 
+            BinaryKey="WixCA" 
+            DllEntry="WixShellExec"
+            Impersonate="yes" />
```
Comment (kc284) [inline WixInstaller/XenCenter.wxs:287]: I believe we should also add `Return="asyncNoWait"` so the installer won't have to wait until the process starts.

File: WixInstaller/XenCenter.wxs [hunk 4]
```diff
@@ -271,6 +271,20 @@
...
         </InstallUISequence>
-        <Icon Id="XenCenterICO" SourceFile="$(env.RepoRoot)\Branding\Images\AppIcon.ico" />
+         <UI>
+            <Publish Dialog="ExitDialog" 
+                Control="Finish" 
+                Event="DoAction" 
+                Value="LaunchApplication">WIXUI_EXITDIALOGOPTIONALCHECKBOX = 1 and NOT Installed
+			</Publish>
+         </UI>
+        <Property Id="WIXUI_EXITDIALOGOPTIONALCHECKBOXTEXT" Value="Launch $(var.BrandProduct)" />
```
Comment (kc284) [inline WixInstaller/XenCenter.wxs:281]: `$(var.BrandConsole)`

Comment (kc284) [reply WixInstaller/XenCenter.wxs:281]: In fact this text will need localization.

**Comment by kc284** [comment]
Regarding the checkbox background issue, because transparency is at the moment only supported for text controls, a possible workaround might be to use a combination of checkbox without text and a text control to achieve the desired result.

**Review by kc284 — CHANGES_REQUESTED** [review]
As above.

**Review by AaronRobson — APPROVED** [review]

**Review by CitrixChris — APPROVED** [review]

**Review by kc284 — APPROVED** [review]

**Pull request merged**